src/lib/initOptions.ts

```diff
@@ -115,11 +116,9 @@ function initOptions(runOptions: RunOptions, { cli }: { cli?: boolean } = {}): O
 
   const format = options.format || []
 
-  // autodetect yarn
-  const files = fs.readdirSync(options.cwd || '.')
-  const autoYarn =
-    !options.packageManager && !options.global && files.includes('yarn.lock') && !files.includes('package-lock.json')
-  if (autoYarn) {
+  const packageManager = determinePackageManager(options)
+
+  if (!options.packageManager && packageManager === 'yarn') {
```

Comment:
I think this can be just `if (packageManager === 'yarn')`?

Reply:
Ah, I remember the original intention now. If the user specifies `options.packageManager` explicitly, there is no need to output the message. ncu aims for minimal design when possible.